zfit/core/basepdf.py

```diff
@@ -0,0 +1,1097 @@
+
+    @property
+    def norm_range(self) -> typing.Union[Range, None]:
+        """Return the current normalization range
+
+        Returns:
+            Range or None: The current normalization range
+
+        """
+        return self._norm_range
+
+    def set_norm_range(self, norm_range: typing.Union[Range, None]):
+        """Fix the normalization range to a certain value. Use with caution!
+
+        It is, in general, better to use either the explicit `norm_range` argument when calling
+        a function or the `temp_norm_range` context manager to set a normalization range for a
+        limited amount of code.
+
+        Args:
+            norm_range ():
+
+        """
+        self._norm_range = convert_to_range(norm_range, dims=Range.FULL)
+        return self
+
+    def _check_input_norm_range(self, norm_range, dims, caller_name="",
+                                none_is_error=False) -> typing.Union[Range, bool]:
+        """If `norm_range` is None, take `self.norm_range`. Convert to :py:class:`Range`
+
+        Args:
+            norm_range (None or Range compatible):
+            dims (tuple(int)):
+            caller_name (str): name of the calling function. Used for exception message.
+            none_is_error (bool): if both `norm_range` and `self.norm_range` are None, the default
+                value is `False` (meaning: no range specified-> no normalization to be done). If
+                this is set to true, two `None` will raise a Value error.
+
+        Returns:
+            Union[Range, False]:
+
+        """
+        if norm_range is None:
+            if self.norm_range is None:
+                if none_is_error:
+                    raise ValueError("Normalization range `norm_range` has to be specified, either by"
+                                     "\na) explicitly passing it to the function {name}"
+                                     "\nb) using the temp_norm_range context manager to temporary set "
+                                     "a default normalization range. Currently, both are None/False."
+                                     "".format(name=caller_name))
+                else:
+                    norm_range = False
+            else:
+                norm_range = self.norm_range
+        return convert_to_range(limits=norm_range, dims=dims)
+
+    @property
+    def dims(self) -> typing.Tuple[int]:
+        """Return the dimensions (a tuple of integers) of the function.
+
+        Returns:
+
+        """
+        return tuple(range(self.n_dims))
+
+    @property
+    def n_dims(self):
+        # TODO: improve n_dims setting
+        return self._n_dims
+
+    @n_dims.setter
+    def n_dims(self, n_dims):
+        self._n_dims = n_dims
+
+    @property
+    def is_extended(self) -> bool:
+        """Flag to tell whether the pdf is extended or not.
+
+        Returns:
+            bool:
+        """
+        return self._yield is not None
+
+    @staticmethod
```

Comment:
Do we want this to be static and so public? I would think this is only used internally, but maybe I'm missing a use case...

Reply:
No, you're right, good catch! Is not needed anymore, removed